src/storage/src/hummock/block_cache.rs

```diff
@@ -38,6 +38,14 @@ pub struct BlockHolder {
 }
 
 impl BlockHolder {
+    pub fn from_ref_block(block: &Block) -> Self {
```

Comment:
It's unsafe but the blockholder generated by `from_ref_block` would be always released before `TableHolder`.

Reply:
If the value returned from `from_ref_block` requires manual control to ensure safety, we should at least mark `from_ref_block` as `unsafe`. Though the current code seems safe, in future development it will be easy to misuse it and cause memory problem. Marking it as `unsafe` at least makes us aware that we will be operating on unsafe code and pay more attention.

A much better solution will be, either making `RefEntry` hold an `Arc<Block>` so that there will not be any unsafe code and no lifetime issue, or adding lifetime in `BlockHolder` so that there is no cost on reference counting, though I think the cost of reference counting can be ignored.

Reply:
I have changed it to `Arc<Block>`